src/Rector/Class_/LatteVarTypesBasedOnPresenterTemplateParametersRector.php

```diff
@@ -0,0 +1,204 @@
+use Rector\Nette\ValueObject\LatteVariableType;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+final class LatteVarTypesBasedOnPresenterTemplateParametersRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Adds latte {varType}s based on presenter $this->template parameters',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+// presenters/SomePresenter.php
+<?php
+
+use Nette\Application\UI\Presenter;
+
+class SomePresenter extends Presenter
+{
+    public function renderDefault(): void
+    {
+        $this->template->title = 'My title';
+        $this->template->count = 123;
+    }
+}
+
+// templates/Some/default.latte
+<h1>{$title}</h1>
+<span class="count">{$count}</span>
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+// presenters/SomePresenter.php
+<?php
+
+use Nette\Application\UI\Presenter;
+
+class SomePresenter extends Presenter
+{
+    public function renderDefault(): void
+    {
+        $this->template->title = 'My title';
+        $this->template->count = 123;
+    }
+}
+
+// templates/Some/default.latte
+{varType string $title}
+{varType int $count}
+
+<h1>{$title}</h1>
+<span class="count">{$count}</span>
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node)
+    {
+        if (! $this->nodeTypeResolver->isObjectType($node, new ObjectType('Nette\Application\UI\Presenter'))) {
+            return null;
+        }
+
+        if ($node->name === null) {
+            return null;
+        }
+
+        $shortClassName = $this->nodeNameResolver->getShortName($node);
+        $presenterName = str_replace('Presenter', '', $shortClassName);
+
+        $actionVarTypes = [];
+        foreach ($node->getMethods() as $method) {
+            $fullActionName = $method->name->name;
+            if (! (str_starts_with($fullActionName, 'action') || str_starts_with($fullActionName, 'render'))) {
+                continue;
+            }
+
+            $actionName = str_replace(['action', 'render'], '', $fullActionName);
+            $actionName = lcfirst($actionName);
+
+            if (! isset($actionVarTypes[$actionName])) {
+                $actionVarTypes[$actionName] = [];
+            }
+
+            $actionVarTypes[$actionName] = array_merge(
+                $actionVarTypes[$actionName],
+                $this->findVarTypesForAction($method)
+            );
+        }
+
+        $this->printVarTypesToTemplateFiles($actionVarTypes, $presenterName);
+
+        return null;
+    }
+
+    /**
+     * @return LatteVariableType[]
+     */
+    private function findVarTypesForAction(ClassMethod $method): array
```

Comment:
Could you extract this to own service? It looks like complex logic

Reply:
It's planned in some future PR. I want to create following implementation for Controls and I will see what is common.